### PR TITLE
Bug - Fix bugs in data diagnosis

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Fix bugs in data diagnosis.

**Major Revision**
- fix package import issue of file_handler
- deal with monitor metrics
- fix typo in output_path

### CODE REVIEW

File: superbench/analyzer/data_diagnosis.py [hunk 1]
```diff
@@ -239,10 +244,10 @@ def run(self, raw_data_file, rule_file, baseline_file, output_dir, output_format
             logger.info('DataDiagnosis: Begin to processe {} nodes'.format(len(self._raw_data_df)))
             data_not_accept_df, label_df = self.run_diagnosis_rules(rule_file, baseline_file)
             logger.info('DataDiagnosis: Processed finished')
-            outpout_path = ''
+            output_path = ''
             if output_format == 'excel':
                 output_path = output_dir + '/diagnosis_summary.xlsx'
-                file_handler.output_excel(self._raw_data_df, data_not_accept_df, outpout_path, self._sb_rules)
+                file_handler.output_excel(self._raw_data_df, data_not_accept_df, output_path, self._sb_rules)
```
Comment (cp5555) [inline superbench/analyzer/data_diagnosis.py:250]: It looks like we miss some UTs here

File: superbench/analyzer/data_diagnosis.py [hunk 2]
```diff
@@ -31,10 +31,15 @@ def _get_metrics_by_benchmarks(self, metrics_list):
...
         benchmarks_metrics = {}
         for metric in metrics_list:
-            benchmark = metric.split('/')[0]
-            if benchmark not in benchmarks_metrics:
-                benchmarks_metrics[benchmark] = set()
-            benchmarks_metrics[benchmark].add(metric)
+            if '/' not in metric:
+                if 'other' not in benchmarks_metrics:
+                    benchmarks_metrics['other'] = set()
+                benchmarks_metrics['other'].add(metric)
```
Comment (cp5555) [inline superbench/analyzer/data_diagnosis.py:37]: Add UT. Besides, why we have `other`?? It's really strange.

File: superbench/analyzer/data_diagnosis.py [hunk 3]
```diff
@@ -31,10 +31,15 @@ def _get_metrics_by_benchmarks(self, metrics_list):
         """
         benchmarks_metrics = {}
         for metric in metrics_list:
-            benchmark = metric.split('/')[0]
-            if benchmark not in benchmarks_metrics:
-                benchmarks_metrics[benchmark] = set()
-            benchmarks_metrics[benchmark].add(metric)
+            if '/' not in metric:
+                if 'monitor' not in benchmarks_metrics:
```
Comment (abuccts) [inline superbench/analyzer/data_diagnosis.py:35]: why not use 'monitor/' prefix for monitor metrics?

File: superbench/analyzer/data_diagnosis.py [hunk 4]
```diff
@@ -239,10 +244,10 @@ def run(self, raw_data_file, rule_file, baseline_file, output_dir, output_format
             logger.info('DataDiagnosis: Begin to processe {} nodes'.format(len(self._raw_data_df)))
```
Comment (abuccts) [inline superbench/analyzer/data_diagnosis.py:244]: processe -> process

File: superbench/analyzer/data_diagnosis.py [hunk 5]
```diff
@@ -239,10 +244,10 @@ def run(self, raw_data_file, rule_file, baseline_file, output_dir, output_format
...
             data_not_accept_df, label_df = self.run_diagnosis_rules(rule_file, baseline_file)
             logger.info('DataDiagnosis: Processed finished')
-            outpout_path = ''
+            output_path = ''
             if output_format == 'excel':
                 output_path = output_dir + '/diagnosis_summary.xlsx'
```
Comment (abuccts) [inline superbench/analyzer/data_diagnosis.py:249]: could you concatenate path instead of string?

File: tests/analyzer/test_data_diagnosis.py [hunk 6]
```diff
@@ -176,3 +187,35 @@ def test_data_diagnosis(self):
             assert ('Category' in line)
             assert ('Defective Details' in line)
             assert ('Index' in line)
+
+    def test_data_diagnosis_run(self):
+        """Test for the run process of rule-based data diagnosis."""
+        test_raw_data = str(Path(__file__).parent.resolve()) + '/test_results.jsonl'
+        test_rule_file = str(Path(__file__).parent.resolve()) + '/test_rules.yaml'
+        test_baseline_file = str(Path(__file__).parent.resolve()) + '/test_baseline.json'
```
Comment (abuccts) [inline tests/analyzer/test_data_diagnosis.py:195]: same

File: tests/analyzer/test_data_diagnosis.py [hunk 7]
```diff
@@ -176,3 +187,35 @@ def test_data_diagnosis(self):
...
+        # Test - output in excel
+        DataDiagnosis().run(
+            test_raw_data, test_rule_file, test_baseline_file, str(Path(__file__).parent.resolve()), 'excel'
```
Comment (abuccts) [inline tests/analyzer/test_data_diagnosis.py:198]: set "self.parent_path = Path(__file__).parent.resolve()" in init if you need to use it many times?

File: tests/analyzer/test_data_diagnosis.py [hunk 8]
```diff
@@ -176,3 +187,35 @@ def test_data_diagnosis(self):
...
+        )
+        excel_file = pd.ExcelFile(self.output_excel_file, engine='openpyxl')
```
Comment (abuccts) [inline tests/analyzer/test_data_diagnosis.py:200]: why not compare the output excel to expected excel directly?

Comment (abuccts) [reply tests/analyzer/test_data_diagnosis.py:200]: `pandas.util.testing.assert_frame_equal`

File: tests/analyzer/test_data_diagnosis.py [hunk 9]
```diff
@@ -176,3 +187,35 @@ def test_data_diagnosis(self):
...
+        raw_data_df = excel_file.parse(data_sheet_name)
+        assert (len(raw_data_df) == 3)
+        data_sheet_name = 'Not Accept'
+        data_not_accept_read_from_excel = excel_file.parse(data_sheet_name)
+        assert (len(data_not_accept_read_from_excel) == 2)
+        assert ('Category' in data_not_accept_read_from_excel)
+        assert ('Defective Details' in data_not_accept_read_from_excel)
+        # Test - output in json
+        DataDiagnosis().run(
+            test_raw_data, test_rule_file, test_baseline_file, str(Path(__file__).parent.resolve()), 'json'
```
Comment (abuccts) [inline tests/analyzer/test_data_diagnosis.py:211]: same

File: tests/analyzer/test_data_diagnosis.py [hunk 10]
```diff
@@ -176,3 +187,35 @@ def test_data_diagnosis(self):
...
+        )
+        assert (Path(self.output_json_file).is_file())
+        with Path(self.output_json_file).open() as f:
+            data_not_accept_read_from_json = f.readlines()
+        assert (len(data_not_accept_read_from_json) == 2)
```
Comment (abuccts) [inline tests/analyzer/test_data_diagnosis.py:216]: why not assertEqual output json string and expected string directly?

File: superbench/analyzer/data_diagnosis.py (hunk 3 above)
Comment (cp5555) [reply superbench/analyzer/data_diagnosis.py:35]: I agree. I think we should fix in the monitor.

File: superbench/analyzer/data_diagnosis.py (hunk 4 above)
Comment (yukirora) [reply superbench/analyzer/data_diagnosis.py]: updated

File: superbench/analyzer/data_diagnosis.py (hunk 5 above)
Comment (yukirora) [reply superbench/analyzer/data_diagnosis.py]: updated

File: tests/analyzer/test_data_diagnosis.py (hunk 8 above)
Comment (yukirora) [reply tests/analyzer/test_data_diagnosis.py:199]: good idea, thanks

File: tests/analyzer/test_data_diagnosis.py (hunk 7 above)
Comment (yukirora) [reply tests/analyzer/test_data_diagnosis.py]: ok

File: tests/analyzer/test_data_diagnosis.py (hunk 10 above)
Comment (yukirora) [reply tests/analyzer/test_data_diagnosis.py]: updated

**Review by cp5555 — APPROVED** [review]

File: superbench/analyzer/data_diagnosis.py (hunk 3 above)
Comment (cp5555) [reply superbench/analyzer/data_diagnosis.py]: Solved.

**Review by abuccts — APPROVED** [review]

**Pull request merged**